tools/serverpod_cli/lib/src/generator/open_api/objects/schema.dart

```diff
@@ -0,0 +1,199 @@
+  assert(type.isMapType,
+      'Use mapToJson only when the typeDefinition is of the MapType.');
+
+  Map<String, dynamic> map = {};
+  map['type'] = SchemaObjectType.object.name;
+  if (type.nullable) map['nullable'] = true;
+
+  // If data is Map<String,int>  use last int from generics as
+  // additionalProperties
+  if (type.generics.isEmpty) return map;
+  var lastType = type.generics.last;
+  if (lastType.isListType) {
+    map['additionalProperties'] = listTypeToJson(lastType, true);
+  } else if (lastType.isMapType) {
+    map['additionalProperties'] = mapTypeToJson(lastType, true);
+  } else if (lastType.isOtherType) {
+    map['additionalProperties'] = otherTypeToJson(lastType, true);
+  } else {
+    map['additionalProperties'] = dartCoreTypeToJson(
+      lastType,
+      true,
+    );
+  }
+
+  return map;
+}
+
+/// Serializes a [TypeDefinition]  dart core type (string,bool,double,int,
+/// BigInt,...) to Json
+Map<String, dynamic> dartCoreTypeToJson(TypeDefinition type,
+    [bool child = false]) {
+  assert(type.toSchemaObjectType != SchemaObjectType.other,
+      'SchemaObjectType should not be other type');
+  assert(
+    type.toSchemaObjectType != SchemaObjectType.array,
+  );
+  assert(
+    type.toSchemaObjectType != SchemaObjectType.object,
+  );
+  Map<String, dynamic> map = {};
+  map['type'] = type.toSchemaObjectType.name;
+  if (type.nullable) map['nullable'] = true;
+  return map;
+}
+
+/// Serializes an [SchemaObjectType.other] to json
+Map<String, dynamic> otherTypeToJson(TypeDefinition type,
+    [bool child = false]) {
+  assert(type.toSchemaObjectType == SchemaObjectType.other,
+      'Use otherTypeToJson only when the SchemaObjectType is other.');
+  Map<String, dynamic> map = {};
+  // Other types are always object.
+  if (!child) map['type'] = SchemaObjectType.object.name;
+  map['\$ref'] =
+      _getRef(type.className == 'dynamic' ? 'AnyValue' : type.className);
+  if (type.nullable && !child) map['nullable'] = true;
+  return map;
+}
+
+/// Serializes a [TypeDefinition] [List] type to Json
+Map<String, dynamic> listTypeToJson(TypeDefinition type, [bool child = false]) {
+  assert(type.isListType,
+      'Use listTypeToJson only when the typeDefinition is of the ListType.');
+  Map<String, dynamic> map = {};
+  map['type'] = SchemaObjectType.array.name;
+
+  map['items'] = {};
+
+  var generic = type.generics;
+  if (generic.isEmpty) {
+    map['items']['\$ref'] = _getRef('AnyValue');
+    return map;
+  }
+  if (generic.first.isListType) {
+    map['items'] = listTypeToJson(generic.first, true);
+    return map;
+  }
+  if (generic.first.isMapType) {
+    map['items'] = mapTypeToJson(generic.first, true);
+    return map;
+  }
+
+  if (generic.first.isOtherType) {
+    map['items'] = otherTypeToJson(generic.first, true);
+    return map;
+  }
+  map['items'] = dartCoreTypeToJson(generic.first, true);
+  return map;
+}
+
+///  Schema object used within [ComponentObject]
```

Comment:
Missing period. Please review all comments to make sure they use correct captialization and grammar.